ssl/internal.h

```diff
@@ -2450,6 +2452,22 @@ struct CERT {
   // is only used as a cache in order to implement “get0” functions that return
   // a non-owning pointer to the certificate chain.
   X509 *x509_leaf = nullptr;
+};
+
+struct CERT {
+  static constexpr bool kAllowUniquePtr = true;
+
+  explicit CERT(const SSL_X509_METHOD *x509_method);
+  ~CERT();
+
+  // cert_privatekey_idx ALWAYS points to an element of the |cert_pkeys|
+  // array. OpenSSL implements this as a pointer, but an index is more
+  // efficient.
+  int cert_privatekey_idx = SSL_PKEY_RSA;
```

Comment:
Does OpenSSL do this for purposes of tracking whether this got initialized to the correct value? E.g. they can check that if this is `nullptr` to see if it wasn't correctly initialized, but here we are default to always pointing at the first index.

Reply:
Don't think there's a particular purpose, I applied the idea since their documentation called it out as a potential improvement. It would be worth doing this right from the start, now that we're reimplementing it.
*  https://github.com/openssl/openssl/blob/OpenSSL_1_1_1-stable/ssl/ssl_local.h#L1879-L1885.

In the original, `cert->key` value is assigned the first index during initialization. There's an expectation that the pointer always exists, so it isn't checked (based on the logic within the commit).
* https://github.com/google/boringssl/commit/d1d8078025cc654d60ee873c88740f8a987a77b7#diff-ea838e3fccacc2f14349aefa3d01ba807d2a2cb218878a93cd0fa227215558a4L151-L161